plugins/out_stackdriver/stackdriver_conf.c

```diff
@@ -281,6 +298,38 @@ struct flb_stackdriver *flb_stackdriver_conf_create(struct flb_output_instance *
         ctx->resource = flb_sds_create(FLB_SDS_RESOURCE_TYPE);
     }
 
+    flb_kv_init(&ctx->labels);
+
+    /*
+     * Load monitored resource labels from config files.
+     * Will only use labels specified for a particular resource type at
+     * https://cloud.google.com/logging/docs/api/v2/resource-list
+    */
+    mk_list_foreach(labels, &ins->properties) {
+        kv = mk_list_entry(labels, struct flb_kv, _head);
+        if (strncasecmp(kv->key, "label.", 6) == 0 &&
+            flb_sds_len(kv->key) > 6) {
+                /* Get label name from config file label identifier*/
+                tmp = strtok(kv->key, ".");
+                /* Only needs do be done twice to get to the first '.' */
+                tmp = strtok(NULL, ".");
+                if (tmp == NULL || strncasecmp(tmp, "label", 5) == 0) {
+                    flb_plg_error(ctx->ins, "bad label '%s'",kv->key);
```

Comment:
Careful — from line 310, you know `kv->key` starts with `"label."`. The `strtok` in line 313 overwrites the `'.'` with `'\0'`, so this error will always print `"label"`.
Why use `strtok` anyway? You're just stripping a prefix of a known length (6) that you've verified in line 310. So it should be sufficient to say `tmp = kv->key + 6`, right?

Reply:
Switch to pointer arithmetic